File: engine/src/main/java/com/bartlomiejpluta/base/engine/world/autotile/model/AutoTileLayout.java
package com.bartlomiejpluta.base.engine.world.autotile.model;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum AutoTileLayout {
   LAYOUT_2X2(2, 2),
   LAYOUT_2X3(2, 3);

   private final int columns;
   private final int rows;
}

File: editor/src/main/kotlin/com/bartlomiejpluta/base/editor/autotile/model/AutoTileLayout.kt
package com.bartlomiejpluta.base.editor.autotile.model

enum class AutoTileLayout(val columns: Int, val rows: Int) {
   LAYOUT_2X2(2, 2),
   LAYOUT_2X3(2, 3);

   override fun toString() = "$columns x $rows"
}